web/src/test/java/org/openmrs/web/filter/update/GZIPFilterTest.java

```diff
@@ -0,0 +1,85 @@
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.web.filter.update;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatcher;
+import org.mockito.Mockito;
+import org.openmrs.GlobalProperty;
+import org.openmrs.api.context.Context;
+import org.openmrs.test.Verifies;
+import org.openmrs.web.filter.GZIPFilter;
+import org.openmrs.web.test.BaseWebContextSensitiveTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+import javax.servlet.FilterChain;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.nio.charset.Charset;
+import java.nio.charset.CharsetDecoder;
+import java.util.zip.GZIPOutputStream;
+
+import static org.mockito.Mockito.mock;
+
+/**
+ * Tests some of the methods on the {@link org.openmrs.web.filter.update.GZIPFilter}
+ */
+public class GZIPFilterTest extends BaseWebContextSensitiveTest {
+	
+	/**
+	 * @see {@link org.openmrs.web.filter.GZIPFilter#doFilterInternal(HttpServletRequest,HttpServletResponse, javax.servlet.FilterChain)}
+	 */
+	@Test
+	@Verifies(value = "zip request and response", method = "performGZIPRequest(HttpServletRequest,HttpServletResponse,FilterChain)")
+	public void zipRequestWrapperTest_shouldReturnTrueIfUnzippedContentReadFromWrapperIsTheSameAsContentBeforeZipping()
+	        throws Exception {
+		GlobalProperty property = new GlobalProperty("gzip.acceptCompressedRequestsForPaths", ".*");
+		Context.getAdministrationService().saveGlobalProperty(property);
+		MockHttpServletRequest req = new MockHttpServletRequest();
+		req.setContextPath("http://gzipservletpath");
+		req.addHeader("Content-encoding", "gzip");
+		ByteArrayOutputStream stream = new ByteArrayOutputStream();
+		GZIPOutputStream gzOutput = new GZIPOutputStream(stream);
+		PrintWriter pwriter = new PrintWriter(gzOutput);
+		pwriter.write("message string");
+		pwriter.flush();
+		gzOutput.finish();
+		req.setContent(stream.toByteArray());
+		MockHttpServletResponse resp = new MockHttpServletResponse();
+		FilterChain fil = mock(FilterChain.class);
+		GZIPFilter gzipFilter = new GZIPFilter();
+		gzipFilter.doFilterInternal(req, resp, fil);
+		final ArgumentCaptor<HttpServletRequest> argumentCaptor = ArgumentCaptor.forClass(HttpServletRequest.class);
+		Mockito.verify(fil).doFilter(argumentCaptor.capture(), Mockito.any(HttpServletResponse.class));
+		HttpServletRequest requestArgument = (HttpServletRequest) argumentCaptor.getValue();
```

Comment:
A few blank lines in this test would make the test more readable :) e.g. after lines 53, 57, 64, 68.